updateHostsFile.py

```diff
@@ -630,7 +646,47 @@ def create_initial_file():
     return merge_file
 
 
-def remove_dups_and_excl(merge_file, exclusion_regexes):
+def compress_file(input_file, target_ip, output_file):
+    """
+    Reduce the file dimension removing non-necessary lines (empty lines and comments) and putting multiple domains in each line.
+    Reducing the number of lines of the file, the parsing under Microsoft Windows is much faster.
+
+    Parameters
+    ----------
+    input_file : file
+        The file object that contains the hostnames that we are reducing.
+    target_ip : str
+        The target IP address.
+    output_file : file
+        The file object that will contain the reduced hostnames.
+    """
+
+    input_file.seek(0)  # reset file pointer
+    write_data(output_file, '\n')
+
+    lines = [target_ip]
+    lines_index = 0
+    for line in input_file.readlines():
+        line = line.decode("UTF-8")
+        if line.startswith('#') or line.startswith('\n'):
+            continue
+
+        if line.startswith(target_ip):
+            l = len(lines[lines_index])
```

Comment:
@stefanopini in the code below, I'm curious why you chose these particular limits for line length?  Why limit at all?

Reply:
I was quite sure Windows had a length limit for each line of the hosts file, so I chose a value to reduce the file size without incur in errors.
But, searching the reference for that, I discovered that the limit doesn't refer to the the length of the line, but to length of the single hostname.
On the contrary, the limit that matters in our case is the number of supported hostnames for each line, fixed to 9. I found that also the issue #49 reports that.
Therefore, the generated hosts file must have at maximum 9 hostnames for each line, regardless of the line length. I've just committed an update with the correct `if` statement and little changes.
I apologize for the dumb error, I should have checked it before. Unfortunately, with my length limits, nearly every line contains less than 9 hosts, so I didn't notice the error.
I'll update the stats previously posted with the new values.